### PR TITLE
fix(aws-amplify-react): the username should not be undefined

### LINKED ISSUE #2930: React SignIn component throws exception if form is empty when submitted
**Describe the bug**
React SignIn component triggers error if form is empty when submitted. We are using the built-in React SignIn component (as part of `withAuthenticator`). It will display the unhelpful error `Cannot read property 'username' of undefined` if submitted without first filling data into the form.

**To Reproduce**
1. Install/configure Auth and use built-in `withAuthenticator` HOC to display SignIn component.
2. Click on the 'Sign In' button without filling out the form.
3. Error `Cannot read property 'username' of undefined` appears in banner

**Expected behavior**
I expect some minor validation of the form before attempting to send data to Cognito. I expect that if username and/or password field are empty when the form is submitted, even on first load, that there will be a notification shown to the user.

**Additional context**
Using the debugger, I can see that values in the SignIn component for username and password are undefined if the form is empty (and has never had input) after clicking "Sign In". See [line 70](https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/SignIn.jsx#L70).

Tracing it down, the issue is that there is no check for undefined before attempting to process the value of `username`. [Line 359](https://github.com/aws-amplify/amplify-js/blob/master/packages/auth/src/Auth.ts#L359) in the Auth package throws the error by trying to access the `username` property of undefined.

Note: If at first, data is typed into the form and then deleted, the values passed to `Auth` are no longer undefined and the form appears to work appropriately. It is only if the form is never interacted with before "Sign In" is pressed that the error is presented.

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
fixes #2930 
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/src/Auth/SignIn.jsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default class SignIn extends AuthPiece {
         // avoid submitting the form
         event.preventDefault();
         
-        const { username, password } = this.inputs;
+        const { username='', password } = this.inputs;
```
Comment (alebiavati) [inline packages/aws-amplify-react/src/Auth/SignIn.jsx]: if the error was `Cannot read property 'username' of undefined`, then this won't fix that error.

The error means that `this.inputs` is undefined.

Comment (powerful23) [reply packages/aws-amplify-react/src/Auth/SignIn.jsx]: In fact the error happens here: https://github.com/aws-amplify/amplify-js/blob/master/packages/auth/src/Auth.ts#L359

The `inputs` variable will always be an object. Please check the `AuthPiece`: https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/AuthPiece.jsx where `inputs` is initialized.

Comment (alebiavati) [reply packages/aws-amplify-react/src/Auth/SignIn.jsx]: Got it, thanks!

Comment (powerful23) [reply packages/aws-amplify-react/src/Auth/SignIn.jsx]: 👍

**Pull request merged**